openbas-api/src/main/java/io/openbas/executors/crowdstrike/service/CrowdStrikeExecutorContextService.java

```diff
@@ -55,48 +57,178 @@ public void launchExecutorSubprocess(
+              action.getCommandEncoded());
+          fromIndex = toIndex;
+          toIndex = Math.min(action.getAgents().size(), fromIndex + paginationLimit);
+          Thread.sleep(1000);
+        }
+      } else {
+        this.crowdStrikeExecutorClient.executeAction(
+            action.getAgents().stream().map(Agent::getId).toList(),
+            action.getScriptName(),
+            action.getCommandEncoded());
+        Thread.sleep(1000);
       }
-      default -> throw new RuntimeException("Unsupported platform: " + platform);
     }
+  }
+
+  private List<CrowdStrikeAction> getWindowsActions(
+      List<Agent> agents, Injector injector, String injectId) {
+    List<CrowdStrikeAction> actions = new ArrayList<>();
+    if (!agents.isEmpty()) {
+      CrowdStrikeAction actionWindows = new CrowdStrikeAction();
+      actionWindows.setScriptName(this.crowdStrikeExecutorConfig.getWindowsScriptName());
+      String implantLocation =
+          "$location="
+              + IMPLANT_LOCATION_WINDOWS
+              + ExecutorHelper.IMPLANT_BASE_NAME
+              + UUID.randomUUID()
+              + "\";md $location -ea 0;[Environment]::CurrentDirectory";
+      Endpoint.PLATFORM_TYPE platform = Endpoint.PLATFORM_TYPE.Windows;
+      // x86_64 by default in the register because CS API doesn't provide the platform architecture
+      String executorCommandKey = platform.name() + "." + Endpoint.PLATFORM_ARCH.x86_64.name();
+      String command = injector.getExecutorCommands().get(executorCommandKey);
+      command = WINDOWS_EXTERNAL_REFERENCE + command;
+      command = replaceArgs(platform, command, injectId, AGENT_ID_VARIABLE);
+      command =
+          command.replaceFirst(
+              "\\$?x=.+location=.+;\\[Environment]::CurrentDirectory",
+              Matcher.quoteReplacement(implantLocation));
+      actionWindows.setCommandEncoded(
+          Base64.getEncoder().encodeToString(command.getBytes(StandardCharsets.UTF_8)));
+      actionWindows.setAgents(agents);
+      actions.add(actionWindows);
+    }
+    return actions;
+  }
 
-    String executorCommandKey = platform.name() + "." + arch.name();
-    String command = injector.getExecutorCommands().get(executorCommandKey);
+  private List<CrowdStrikeAction> setLinuxActions(
+      List<Agent> agents, Injector injector, String injectId) {
+    List<CrowdStrikeAction> actions = new ArrayList<>();
+    if (!agents.isEmpty()) {
+      CrowdStrikeAction actionLinux = new CrowdStrikeAction();
+      actionLinux.setScriptName(this.crowdStrikeExecutorConfig.getUnixScriptName());
+      actionLinux.setCommandEncoded(
+          getUnixCommand(
+              Endpoint.PLATFORM_TYPE.Linux, injector, injectId, LINUX_EXTERNAL_REFERENCE));
+      actionLinux.setAgents(agents);
+      actions.add(actionLinux);
+    }
+    return actions;
+  }
 
-    command = replaceArgs(platform, command, inject.getId(), agent.getId());
+  private List<CrowdStrikeAction> setMacActions(
```

Comment:
nitpick: Possible to rename to setMacOSActions? It could be mixed up with Mac adress otherwise (for readers not paying attention).